src/inventory_client/inventory_client.go

```diff
@@ -41,10 +42,18 @@ type EnabledHostData struct {
 	Host      *models.Host
 }
 
-func CreateInventoryClient(clusterId string, host string, port int, logger *logrus.Logger) *inventoryClient {
+// PullSecretWriter provides a header based pullsecret auth info writer
+func PullSecretWriter(token string) runtime.ClientAuthInfoWriter {
+	return runtime.ClientAuthInfoWriterFunc(func(r runtime.ClientRequest, _ strfmt.Registry) error {
+		return r.SetHeaderParam("X-Secret-Key", token)
```

Comment:
This constant must be aligned with what bm-inventory expects.
I think this entire function should be in the /pkg directory of bm-inventory.

It will also be used by assisted-controller, no?

Reply:
sounds good. It will be relevant for all clients right?

Reply:
OK. I will add it to bm-inventory after User Auth is merged

Reply:
Done